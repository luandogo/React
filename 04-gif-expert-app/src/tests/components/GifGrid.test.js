import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Tests in <GifGrid />', () => {

    const category = 'One Punch Man'

    test('should be displayed correctly.', () => {
        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });
        const wrapper = shallow( <GifGrid category={ category }/> );
        expect( wrapper ).toMatchSnapshot();
        
    });

    test('should show items when images are loaded useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/something/something.jpg',
            title: 'Something'
        },{
            id: 'ABC123',
            url: 'https://localhost/something/something.jpg',
            title: 'Something'
        }];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
        
        
        const wrapper = shallow( <GifGrid category='{ category }'/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('p').exists() ).toBe(false);        
        expect( wrapper.find('GifGridItem').length ).toBe( gifs.length );
    });
    
    
 
});