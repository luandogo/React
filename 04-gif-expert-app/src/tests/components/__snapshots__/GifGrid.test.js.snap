// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifGrid /> should be displayed correctly. 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch Man
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
     Loading 
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Tests in <GifGrid /> should show items when images are loaded useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    { category }
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Something"
      url="https://localhost/something/something.jpg"
    />
    <GifGridItem
      id="ABC123"
      key="ABC123"
      title="Something"
      url="https://localhost/something/something.jpg"
    />
  </div>
</Fragment>
`;
